broker-daemon/state-machines/plugins/persistence.js

```diff
@@ -1,74 +1,76 @@
-   */
+/**
+ * Unique key accessor for saving and inflating state machines
+ * @typedef {(Function|string)} StateMachinePersistenceKeyAccessor
+ *     If a string is passed, the accessor will act as an accessor on property specified by the string.
+ * @param {string}      key   - Key being used to inflate the state machine for which this accessor is being set. Will be undefined if being used as a getter.
+ * @param {object}      value - Object being used to inflate the state machine for which this accessor is being set. Will be undefined if being used as a getter.
+ */
 
-  /**
-   * Unique key accessor for saving and inflating state machines
-   * @typedef {(Function|String)} StateMachinePersistence~KeyAccessor
-   * If a string is passed, the accessor will act as an accessor on property specified by the string.
-   * @this {StateMachine}       State machine instance being inflated
-   * @param {String}      key   Key being used to inflate the state machine for which this accessor is being set. Will be undefined if being used as a getter.
-   * @param {Object}      value Object being used to inflate the state machine for which this accessor is being set. Will be undefined if being used as a getter.
-   */
+/** @typedef {{[k: string]: StateMachinePersistenceFieldAccessor}} AdditionalFields */
 
+/**
+ * @class Store state machine data and state using a LevelUp compatible store
+ */
+class StateMachinePersistence extends StateMachinePlugin {
   /**
    * Set up the persistence plugin with user-defined attributes to save additional fields and avoid naming conflicts
-   * @param {Object} options
-   * @param  {StateMachinePersistence~KeyAccessor}                   options.key              - Name of the property on the state machine instance with the unique key, or a function that derives the key from the instance
-   * @param  {Object<string, StateMachinePersistence~FieldAccessor>} options.additionalFields - List of additional properties on the instance to be persisted
-   * @param  {string}                                                options.storeName        - Name of the property on the state machine where the StateMachinePersistence~Store is located
+   * @param {object} options
+   * @param {StateMachinePersistenceKeyAccessor} options.key - Name of the property on the state machine instance with the unique key, or a function that derives the key from the instance
+   * @param {AdditionalFields} options.additionalFields - List of additional properties on the instance to be persisted
+   * @param {string} [options.storeName] - Name of the property on the state machine where the StateMachinePersistenceStore is located
    */
-  constructor ({ key = 'id', additionalFields = {}, storeName = 'store' } = {}) {
+  constructor ({ key = 'id', additionalFields = {}, storeName = 'store' } =
+  { key: 'id', additionalFields: {}, storeName: 'store' }) {
```

Comment:
this seems weird

Reply:
Yeah, should we just make it always require an argument?

Reply:
yes that's a good idea